internal/operands/data-sources/reconcile.go

```diff
@@ -330,6 +343,48 @@ func addToCronMap(cronMap map[client.ObjectKey]*cdiv1beta1.DataImportCron, cron
 	}] = cron
 }
 
+// addDataSourceReferenceForCrons adds DataSource references for custom DataImportCron templates.
+// The SSP object can contain DataImportCron templates that don't have a common template defined.
+func addDataSourceReferenceForCrons(dataSourceInfos []dataSourceInfo, sspSpec *ssp.SSPSpec) ([]dataSourceInfo, error) {
```

Comment:
would be please possible to divide this function into smaller functions?

Reply:
I've extracted the inner loop into a function, is it enough? The other are just conditions when to skip the current iteration.